### PR TITLE
Improve code style in cluster

### PR DESCRIPTION
**Pull request by jihuayu**
I haven't modified any specific logic. Only tidy codes to make Sonarcloud happy.

https://sonarcloud.io/project/issues?fileUuids=AYuMX0sBEoUnDAVyI8Dx&resolved=false&id=apache_kvrocks

### CODE REVIEW

File: src/cluster/cluster.h [hunk 1]
```diff
@@ -97,18 +97,18 @@ class Cluster {
...
   Server *srv_;
   std::vector<std::string> binds_;
   int port_;
-  int size_;
-  int64_t version_;
+  int size_ = 0;
+  int64_t version_ = -1;
   std::string myid_;
-  std::shared_ptr<ClusterNode> myself_;
+  std::shared_ptr<ClusterNode> myself_ = nullptr;
```
Comment (PragmaTwice) [inline src/cluster/cluster.h]: It's useless.

File: src/cluster/cluster.cc [hunk 2]
```diff
@@ -567,10 +565,10 @@ std::string Cluster::genNodesDescription() {
   return nodes_desc;
 }
 
-std::map<std::string, std::string> Cluster::getClusterNodeSlots() const {
+std::map<std::string, std::string, std::less<>> Cluster::getClusterNodeSlots() const {
```
Comment (PragmaTwice) [inline src/cluster/cluster.cc:570]: I don't think we should specify the compare type explicitly.

If it reported by sonarcloud, please suppress it.

File: src/cluster/cluster.h [hunk 3]
```diff
@@ -97,18 +97,18 @@ class Cluster {
  private:
   std::string getNodeIDBySlot(int slot) const;
   std::string genNodesDescription();
-  std::string genNodesInfo();
-  std::map<std::string, std::string> getClusterNodeSlots() const;
+  std::string genNodesInfo() const;
+  std::map<std::string, std::string, std::less<>> getClusterNodeSlots() const;
```
Comment (PragmaTwice) [inline src/cluster/cluster.h:101]: ditto

File: src/cluster/cluster.cc (hunk 2 above)
Comment (jihuayu) [reply src/cluster/cluster.cc:570]: Yes, It is according to https://sonarcloud.io/organizations/apache/rules?open=cpp%3AS6045&rule_key=cpp%3AS6045.
We can disable it in [Quality Profiles](https://sonarcloud.io/project/quality_profiles?id=apache_kvrocks)

Comment (mapleFU) [reply src/cluster/cluster.cc:570]: Maybe this is for find the string_view?

Comment (PragmaTwice) [reply src/cluster/cluster.cc:570]: Hmm I got the motivation. Sorry we can keep them for key type = std::string.

File: src/cluster/cluster.cc [hunk 4]
```diff
@@ -37,11 +37,11 @@
 #include "time_util.h"
 
 ClusterNode::ClusterNode(std::string id, std::string host, int port, int role, std::string master_id,
-                         std::bitset<kClusterSlots> slots)
+                         std::bitset<kClusterSlots> const &slots)
     : id(std::move(id)), host(std::move(host)), port(port), role(role), master_id(std::move(master_id)), slots(slots) {}
```
Comment (mapleFU) [inline src/cluster/cluster.cc]: what about using std::move?

Comment (PragmaTwice) [reply src/cluster/cluster.cc]: I don't think it will work. The data in bitset is on the stack intead of heap.

File: src/cluster/cluster.cc [hunk 5]
```diff
@@ -53,10 +53,10 @@ Cluster::Cluster(Server *srv, std::vector<std::string> binds, int port)
 // cluster data, so these commands should be executed exclusively, and ReadWriteLock
 // also can guarantee accessing data is safe.
 bool Cluster::SubCommandIsExecExclusive(const std::string &subcommand) {
-  for (auto v : {"setnodes", "setnodeid", "setslot", "import", "reset"}) {
-    if (util::EqualICase(v, subcommand)) return true;
-  }
-  return false;
+  static std::vector<std::string> values = {"setnodes", "setnodeid", "setslot", "import", "reset"};
```
Comment (PragmaTwice) [inline src/cluster/cluster.cc]: This `static` will bring a lock to this function. Please remove it.

Comment (PragmaTwice) [reply src/cluster/cluster.cc]: And also we don't need a vector here, array or init list is enough.

File: src/cluster/cluster.cc [hunk 6]
```diff
@@ -37,11 +39,11 @@
 #include "time_util.h"
 
 ClusterNode::ClusterNode(std::string id, std::string host, int port, int role, std::string master_id,
-                         std::bitset<kClusterSlots> slots)
+                         std::bitset<kClusterSlots> const &slots)
```
Comment (torwig) [inline src/cluster/cluster.cc]: I think in this project we usually put the `const` qualifier before the type:
`const std::bitset<kClusterSlots> &slots`

File: src/cluster/cluster.cc [hunk 7]
```diff
@@ -53,9 +55,9 @@ Cluster::Cluster(Server *srv, std::vector<std::string> binds, int port)
 // cluster data, so these commands should be executed exclusively, and ReadWriteLock
 // also can guarantee accessing data is safe.
 bool Cluster::SubCommandIsExecExclusive(const std::string &subcommand) {
-  static std::vector<std::string> values = {"setnodes", "setnodeid", "setslot", "import", "reset"};
+  std::array<std::string, 5> subcommands = {"setnodes", "setnodeid", "setslot", "import", "reset"};
```
Comment (PragmaTwice) [inline src/cluster/cluster.cc]:
```suggestion
  std::string_view subcommands[] = {"setnodes", "setnodeid", "setslot", "import", "reset"};
```

File: src/cluster/cluster.cc [hunk 8]
```diff
@@ -53,9 +55,9 @@ Cluster::Cluster(Server *srv, std::vector<std::string> binds, int port)
...
 
-  return std::any_of(values.begin(), values.end(),
+  return std::any_of(subcommands.begin(), subcommands.end(),
```
Comment (PragmaTwice) [inline src/cluster/cluster.cc]:
```suggestion
  return std::any_of(std::begin(subcommands), std::end(subcommands),
```

**Review by git-hulk — APPROVED** [review]
LGTM

**Review by mapleFU — APPROVED** [review]

File: src/cluster/cluster.cc [hunk 9]
```diff
@@ -53,10 +55,10 @@ Cluster::Cluster(Server *srv, std::vector<std::string> binds, int port)
...
 // also can guarantee accessing data is safe.
 bool Cluster::SubCommandIsExecExclusive(const std::string &subcommand) {
-  for (auto v : {"setnodes", "setnodeid", "setslot", "import", "reset"}) {
-    if (util::EqualICase(v, subcommand)) return true;
-  }
-  return false;
+  std::array subcommands = {"setnodes", "setnodeid", "setslot", "import", "reset"};
+
+  return std::any_of(std::begin(subcommands), std::end(subcommands),
+                     [&subcommand](const std::string &val) { return util::EqualICase(val, subcommand); });
```
Comment (PragmaTwice) [inline src/cluster/cluster.cc:61]: The parameter type should be `string_view` or `auto`.

Comment (jihuayu) [reply src/cluster/cluster.cc:61]: sorry I will fix it

**Pull request merged**